### PR TITLE
feat(kafka): add --wait flag to perform synchronous Kafka creation

### PR DESCRIPTION
**Pull request by craicoverflow**
Closes https://github.com/redhat-developer/app-services-cli/issues/604

### Verification Steps

1. Run `rhoas kafka create my-kafka -w`
2. The terminal will output the current status of the Kafka instance. Once it is complete it will print out the Kafka configuration object.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [x] Documentation added for the feature
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/cmd/kafka/create/create.go [hunk 1]
```diff
@@ -174,48 +180,70 @@ func runCreate(opts *Options) error {
...
+				return err
+			}
+		}
+		s.Stop()
+		logger.Info()
+		logger.Info()
+		logger.Info(opts.localizer.MustLocalize("kafka.create.info.successSync", localize.NewEntry("Name", response.GetName())))
+	}
+
+	switch opts.outputFormat {
```
Comment (wtrocki) [inline pkg/cmd/kafka/create/create.go:233]: FYI There is helper for that in dump package. We no longer need this boilerplate code

File: pkg/iostreams/iostreams.go [hunk 2]
```diff
@@ -88,6 +90,11 @@ func (s *IOStreams) IsSSHSession() bool {
 	return hasClient || hasTTY
 }
 
+// NewSpinner returns a new spinner progress bar to be used in synchronous commands
+func (s *IOStreams) NewSpinner() *spinner.Spinner {
```
Comment (wtrocki) [inline pkg/iostreams/iostreams.go:94]: CC @ankithans

File: pkg/cmd/kafka/create/create.go (hunk 1 above)
Comment (craicoverflow) [reply pkg/cmd/kafka/create/create.go:233]: Yeah, I'm aware. Would like that done another time (this switch already existed).

File: pkg/cmd/kafka/create/create.go [hunk 3]
```diff
@@ -174,48 +180,70 @@ func runCreate(opts *Options) error {
...
 
+	if opts.wait {
+		logger.Debug("--wait flag is enabled, waiting for Kafka to finish creating")
+		s := opts.IO.NewSpinner()
+		s.Suffix = fmt.Sprintf(" %v", opts.localizer.MustLocalize("kafka.create.log.info.creatingKafka", localize.NewEntry("Name", opts.name)))
+		s.Start()
+
+		for svcstatus.IsCreating(response.GetStatus()) {
+			time.Sleep(cmdutil.DefaultPollTime)
+			s.Suffix = " " + opts.localizer.MustLocalize("kafka.create.log.info.creationInProgress", localize.NewEntry("Name", response.GetName()), localize.NewEntry("Status", color.Info(response.GetStatus())))
```
Comment (wtrocki) [inline pkg/cmd/kafka/create/create.go:220]:
```suggestion
			s.Suffix = " " + opts.localizer.MustLocalize("kafka.create.log.info.creationInProgress", 
			localize.NewEntry("Name", response.GetName()), 
			localize.NewEntry("Status", color.Info(response.GetStatus())))
```

File: pkg/cmd/kafka/create/create.go [hunk 4]
```diff
@@ -174,48 +180,70 @@ func runCreate(opts *Options) error {
...
+			response, httpRes, err = api.Kafka().GetKafkaById(context.Background(), response.GetId()).Execute()
+			logger.Debug("Checking Kafka status:", response.GetStatus())
+			if err != nil {
+				return err
+			}
+		}
+		s.Stop()
+		logger.Info()
+		logger.Info()
```
Comment (wtrocki) [inline pkg/cmd/kafka/create/create.go:229]:
```suggestion
		logger.Info("\n\n")
```

File: pkg/cmd/kafka/create/create.go (hunk 1 above)
Comment (wtrocki) [reply pkg/cmd/kafka/create/create.go:233]: Sorry for offtopic.

File: pkg/localize/locales/en/cmd/kafka_create.en.toml [hunk 5]
```diff
@@ -38,13 +38,23 @@ one = 'Cloud Provider Region ID'
 [kafka.create.flag.autoUse.description]
 one = 'Set the new Kafka instance to the current instance'
 
+[kafka.create.flag.wait.description]
+one = 'Wait for the Kafka instance to finish creating'
```
Comment (bhardesty) [inline pkg/localize/locales/en/cmd/kafka_create.en.toml:42]:
```suggestion
one = 'Wait until the Kafka instance is created'
```

**Review by rkpattnaik780 — COMMENTED** [review]
Should we put a message if user cancels creation with `Ctrl +C`. It should imply that creation is still going on (and not cancelled) and user can check status.

Screenshot from 2021-08-24 17-49-28

Based on intuition, I would expect the create operation to be entirely cancelled if I do a `Ctrl +C`, what do you think?

**Comment by craicoverflow** [comment]
> Should we put a message if user cancels creation with `Ctrl +C`. It should imply that creation is still going on (and not cancelled) and user can check status.
> 
> Screenshot from 2021-08-24 17-49-28
> 
> Based on intuition, I would expect the create operation to be entirely cancelled if I do a `Ctrl +C`, what do you think?

Yeah I like that. Will look into it.

**Review by wtrocki — APPROVED** [review]

**Comment by craicoverflow** [comment]
I have added logic to detect a SIGINT and shows an appropriate message. Mind checking that?

**Review by rkpattnaik780 — APPROVED** [review]
Looks great! Just a typo.

File: pkg/localize/locales/en/cmd/kafka_create.en.toml [hunk 6]
```diff
@@ -38,13 +38,26 @@ one = 'Cloud Provider Region ID'
...
+description = 'Message when Kafka instance is being created'
+one = 'Kafka instance "{{.Name}}" is being created. Current status: {{.Status}}.'
+
+[kafka.create.info.successAsync]
 description = 'Message to display when instance has been created'
-one = 'Kafka instance "{{.Name}}" is being provisioned. You can monitor its progress by running "rhoas status".'
+one = 'Kafka instance "{{.Name}}" is being created. To monitor its status run "rhoas status".'
+
+[kafka.create.log.info.creatingKafkaSyncSigint]
+one = 'Your Kafka instance is being created in the background. To monitor its status run "rhoas status" or "rhoas describe".'
```
Comment (rkpattnaik780) [inline pkg/localize/locales/en/cmd/kafka_create.en.toml:57]: "rhoas describe" -> "rhoas kafka describe"

**Pull request merged**